R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+#' Get dependencies for all elements in lock
+get_ordered_dependencies <- function(lock) {
+   cat(crayon::blue(paste("Retrieving dependency info...")))
+   deps <- get_dependencies_for_list(lock, lock, list())
+   cat("\n")
+   deps
+}
+
+#' Recursive function to take a list and lock and extract dependencies, sorting
+#' along the way using the combine_dependencies function.
+get_dependencies_for_list <- function(master_list, lock, previously_parsed_deps) {
+  current_dependencies <- master_list
+  for (i in 1:length(master_list)) {
+    package <- master_list[[i]]
+    previously_parsed_loc <- which_previously_parsed(
+        package, previously_parsed_deps)
```

Comment:
Can you put more of things like this throughout on one line? It would make this block a lot more readable.